### ISSUE TITLE
Task Management

### ISSUE THREAD

**Issue by IvanPombo** [inductiva/inductiva#870]
**TLDR:** Launching simulations with the API is now super easy. Managing them afterwards not so much.

To create a great user experience, the simulation/task management needs to be improved. The main goal is to provide users with the necessary tools and avoid the necessity to write great amounts of code for features that could be done via one line with **Inductiva API**. 

The user workflow needs to be clean and without clutter they should be able to manage their simulations, generate datasets and verify everything is according to their expectations.

The purpose of this issue is to start gathering the several problems and constraints to break down into several issues further ahead.

## Workflows

There are two main workflows when using the API, which are mostly perpendicular but the backend can be similar.
**1 -** Launching several simulations in parallel to generate datasets or explore a design space;
**2 -** Run a single simulation at a time to tweak the product or setup the configuration.

Do to these two approaches, the API needs to offer viable options to deal with both. Currently, the workflow for the point 1 is the most limited.

#### Breakdown of **1**

- [ ] Simulation Project/Experiment: For the generation of datasets, the management of all of the simulation tasks can become cleaner with the concept of a Project/Experiment class. The purpose is to allow tasks to be attached to the Project and create management tools to iterate over them. Methods like, fetch failed tasks from project, or, download outputs of all tasks, can become one-liners with this concept. Within a project, all tasks will be inside the same bucket in the cloud.

This is the main breakdown, then other things surge that fit both 1 and 2, and can make the life of users much simpler.

#### Features and Problems

- [ ] Executer-failed status needs to be broken down into various specific errors, so that users can infer what actually happened. If its their fault then they should be completely aware.
- [ ] Task Resubmission: After a simulation finishes with an error, like executer-failed, users should be able to re-launch their simulations to a new or the same set of machines.
- [ ] Mix between stderr and stdout files. These is occurring on at least one simulator - **FDS** - further debug will be required to understand for other simulators.
- [ ] Price attribute for each task, after the simulation finishes.
- [ ] Pre- and Post- processing methods of tasks from data already in the bucket.
- [ ] Documentation of tasks should improve to be more sequential and not just a ramble of things.
- [ ] What are the priority things to manage tasks that should be on the CLI and console?
- [ ] Should users receive an email after their simulations finish?
- [ ] Metrics of the computational resources are super important to inform the user if they are using their resources to the full potential.
- [ ] Saving the terminal logs in the user side to be more accessible for debugging.
